src/utils/pluginDriver.ts

```diff
@@ -39,22 +41,49 @@ export type HookContext = (context: PluginContext, plugin?: Plugin) => PluginCon
 export function createPluginDriver(
 	graph: Graph,
 	options: InputOptions,
+	pluginCache: Record<string, SerialisablePluginCache>,
 	watcher?: Watcher
 ): PluginDriver {
 	const plugins = [...(options.plugins || []), getRollupDefaultPlugin(options)];
-	const { emitAsset, getAssetFileName, setAssetSource } = createAssetPluginHooks(graph.assetsById);
+	const { emitAsset, getAssetFileName, setAssetSource } = createAssetPluginHooks(
+		graph.assetsById,
+		graph.watchFiles
+	);
+	const existingPluginKeys: string[] = [];
 
 	let hasLoadersOrTransforms = false;
 
 	const pluginContexts = plugins.map(plugin => {
+		let cacheable = true;
+		if (typeof plugin.cacheKey === 'string') {
```

Comment:
This is an empty condition. Is it necessary?

Reply:
It's a ruling-out case for the others to avoid a repeated check. Could be done better certainly, but that was the logic that made sense to me.

Reply:
I've "conventionalized" it :P